server/src/error.rs

```diff
@@ -0,0 +1,63 @@
+use std::convert::Infallible;
+use warp::{http::StatusCode, Rejection, Reply};
+
+pub(crate) mod context;
+
+pub(crate) struct ErrReport(pub(crate) eyre::ErrReport<context::Context>);
```

Comment:
What's the `ErrReport` for?

Reply:
Its a wrapper so I can impl Reject for eyre::ErrReport which is a requirement for errors returned through warp

![Screenshot from 2020-04-07 15-22-04](https://user-images.githubusercontent.com/1993852/78725127-a935b500-78e3-11ea-80a7-5af6b5c49147.png)

Reply:
Cool, maybe it would be helpful to have some motivational comment on the error handling code?